src/FlagProvider.tsx

```diff
@@ -107,7 +107,7 @@ const FlagProvider: FC<PropsWithChildren<IFlagProvider>> = ({
     () => ({
       on: ((event, callback, ctx) => client.current.on(event, callback, ctx)) as IFlagContextValue['on'],
       off: ((event, callback) => client.current.off(event, callback)) as IFlagContextValue['off'],
-      updateContext: async (context) => await client.current.updateContext(context),
+      updateContext: (context) => client.current.updateContext(context),
```

Comment:
Hi @pr28jat I can't spot any difference in this suggestion? Can you try https://github.com/Unleash/proxy-client-react/pull/174#discussion_r1706667737 ?

Reply:
Hi @gastonfournier , sorry for delay. I've updated it as suggested. Thanks!

Reply:
Thanks! The tests are failing though, I've tried locally and they also fail. Currently, this is not a priority to us because it's really old. If you manage to fix the tests in a simple way, we will consider merging this, but we don't want to fix support for an old versions if this risks correctness of our SDK

Reply:
I see the test-case expects the value to be a promise, which if it's not necessary then I can update the unit test as well.

>  If you manage to fix the tests in a simple way, we will consider merging this, but we don't want to fix support for an old versions if this risks correctness of our SDK

That's fine for me @gastonfournier , for now I'm using `patch-package` to modify the code on `node_modules` upon install.

Reply:
I just looked at the test, modified it locally, and all tests go green. I'm curious why the previous behavior was ok... or if it was intentional 2 years ago: https://github.com/iamchanii/proxy-client-react/commit/5ca3a02360c8ea4e6e1f8f0d5cc4770a8bde7e4a#diff-465cbe85eff4124dbbe0c763221ff0ae31086b17e85a53b1dd244397a45da94dR183

In any case both `[object Promise]` and `undefined` feel wrong, but I think it doesn't hurt to change that (because everything works and that seems to be an edge case).

So something like this: https://github.com/chenxeed/proxy-client-react/pull/1 would do it. And then I can probably merge this one

Reply:
Cool, I still don't have permissions to merge it because it's in your repository, would you mind merging it into this PR, I hope that solves the problem :pray:

Reply:
I'm sorry @gastonfournier I just saw it and just merged it today. Thanks again for the PR!